test/libsolidity/syntaxTests/array/concat/bytes_concat_valid_type_literal.sol

```diff
@@ -0,0 +1,6 @@
+contract C {
+    function f() public pure {
+        bytes.concat(0, 0.0, -0, 0e10, 0e-10, 0x00, hex"00", unicode"abc", "abc", (0));
```

Comment:
Why do we want to allow things like `0.0` and `-1` for `bytes.concat`?

Reply:
Don’t allow any changes all the illegal from the hacks have been notified to authorities. So whoever keep deleting and changes my files and keep adding on new one are responsible

Reply:
Probably no one thought to explicitly forbid them :) They are all rational number literals with a value of zero and are implicitly convertible to fixed bytes so the type checker accepts them. Here I'm just making it not crash with the stuff that's already allowed so I think forbidding it is out of the scope of this PR.

Actually, the question should probably be why they are implicitly convertible to fixed bytes in the first place. If they are, I don't see a reason to disallow them so maybe we should rather get rid of the implicit conversion?

Reply:
Should definitely get rid of the implicit conversion :)

Reply:
I created an issue: #11582.